gadgets/trace_exec/test/unit/trace_exec_test.go

```diff
@@ -0,0 +1,203 @@
+			additionalValidation: func(t *testing.T, info *utilstest.RunnerInfo, event ExpectedTraceExecEvent) {
+				require.Equal(t, "ENOENT", event.Error)
+			},
+		},
+		"successful_exec_from_thread": {
+			runnerConfig:  &utilstest.RunnerConfig{},
+			argv:          []string{"/bin/echo", "hello", "world"},
+			runFromThread: true,
+		},
+		"failed_exec_from_thread": {
+			runnerConfig:  &utilstest.RunnerConfig{},
+			argv:          []string{"/bin/meowmeow", "hello", "world"},
+			runFromThread: true,
+			additionalValidation: func(t *testing.T, info *utilstest.RunnerInfo, event ExpectedTraceExecEvent) {
+				require.Equal(t, "ENOENT", event.Error)
+			},
+		},
+	}
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			runner := utilstest.NewRunnerWithTest(t, testCase.runnerConfig)
+			var mntnsFilterMap *ebpf.Map
+			if testCase.mntnsFilterMap == nil {
+				// by default we use the mount namespace of the runner
+				mntnsFilterMap = utilstest.CreateMntNsFilterMap(t, runner.Info.MountNsID)
+			} else {
+				mntnsFilterMap = testCase.mntnsFilterMap(runner.Info)
+			}
+			onGadgetRun := func(gadgetCtx operators.GadgetContext) error {
+				utilstest.RunWithRunner(t, runner, func() error {
+					if testCase.runFromThread {
+						generateEventFromThread(testCase.argv)
+					} else {
+						os.StartProcess(testCase.argv[0], testCase.argv, &os.ProcAttr{})
+					}
+					return nil
+				})
+				return nil
+			}
+			opts := gadgetrunner.GadgetRunnerOpts[ExpectedTraceExecEvent]{
+				Image:   "trace_exec",
+				Timeout: 5 * time.Second,
+				ParamValues: api.ParamValues{
+					"operator.oci.ebpf.ignore-failed": "false",
+				},
+				MntnsFilterMap: mntnsFilterMap,
+				OnGadgetRun:    onGadgetRun,
+			}
+			gadgetRunner := gadgetrunner.NewGadgetRunner(t, opts)
+
+			gadgetRunner.RunGadget()
+
+			if testCase.eventsBlocked {
+				require.Len(t, gadgetRunner.CapturedEvents, 0, "Expected no events but got %d", len(gadgetRunner.CapturedEvents))
+				return
+			}
+
+			var ev ExpectedTraceExecEvent
+			if testCase.runFromThread {
+				// 1 extra event to run the python script
+				require.Len(t, gadgetRunner.CapturedEvents, 2, "Expected 2 events but got %d", len(gadgetRunner.CapturedEvents))
+				require.Contains(t, gadgetRunner.CapturedEvents[0].Args, "/usr/bin/python3 -c")
+				ev = gadgetRunner.CapturedEvents[1]
+			} else {
+				require.Len(t, gadgetRunner.CapturedEvents, 1, "Expected 1 event but got %d", len(gadgetRunner.CapturedEvents))
+				ev = gadgetRunner.CapturedEvents[0]
+			}
+
+			// can only capture TOTAL_ARGS_SIZE ~ 20 arguments
+			if len(testCase.argv) > 20 {
+				testCase.argv = testCase.argv[:20]
+			}
+			expectedArgs := strings.Join(testCase.argv, " ")
+
+			require.Equal(t, expectedArgs, ev.Args, "Expected Args %q, got %q", expectedArgs, ev.Args)
+
+			if testCase.additionalValidation != nil {
+				testCase.additionalValidation(t, runner.Info, ev)
+			}
```

Comment:
I understand here some generic logic is handled to avoid repeating that again on each test, however I think there is too much specific logic. Is it possible to move it inside each test additionalValidation?